indico/htdocs/js/indico/jquery/permissions_widget.js

```diff
@@ -30,30 +30,47 @@
             this.$dataField.val(JSON.stringify(this.data));
         },
         _renderLabel: function(principal) {
-            var $labelBox = $('<div class="label-box">');
+            var $labelBox = $('<div>', {class: 'label-box'});
             if (principal.type === 'role') {
-                var $code = $('<span class="role-code">').text(principal.code)
-                    .css({'border-color': '#' + principal.color, 'color': '#' + principal.color});
-                var $text = $('<span class="text-normal">').text(principal.name);
-                return $labelBox.append($('<span class="flexrow f-a-center">').append($code).append($text));
+                var $text = $('<span>', {class: 'text-normal', text: principal.name});
+                var $code = $('<span>', {
+                    class: 'role-code',
+                    text: principal.code,
+                    css: {
+                        'border-color': '#' + principal.color,
+                        'color': '#' + principal.color
+                    }
+                });
+
+                return $labelBox.append($('<span>', {class: 'flexrow f-a-center'}).append($code).append($text));
             } else {
                 var iconClass = principal.type === 'user' ? 'icon-user' : 'icon-users';
                 var text = principal.type === 'user' ? principal.name : principal.id;
-                return $labelBox.append($('<span class="label-icon text-normal">').addClass(iconClass).text(text));
+                return $labelBox.append($('<span>', {class: 'label-icon text-normal ' + iconClass, text: text}));
```

Comment:
Is it really better to add a class this way? Imagine you have to add two or more classes, how would you do?
IMO that space after `text-normal` is quite ugly. More opinions @ThiefMaster @pferreir @meluru?

Reply:
Surrounding space in class attributes is fine. It's read by browsers, not humans ;) We do it all the time in Jinja templates since avoiding it would make it less readable to humans.

Anyway, generating markup from plain JS/jQuery is always ugly. No strong opinion here...